src/components/ha-spinner.ts

```diff
@@ -0,0 +1,570 @@
+import {
+  LitElement,
+  TemplateResult,
+  css,
+  property,
+  PropertyValues,
+  html,
+  customElement,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+
+// Adapted from polymer element https://github.com/PolymerElements/ha-spinner
+/**
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at
+http://polymer.github.io/LICENSE.txt The complete set of authors may be found at
+http://polymer.github.io/AUTHORS.txt The complete set of contributors may be
+found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by Google as
+part of the polymer project is also subject to an additional IP rights grant
+found at http://polymer.github.io/PATENTS.txt
+*/
+
+@customElement("ha-spinner")
+class HaSpinner extends LitElement {
+  @property({ type: Boolean })
+  public active = false;
+
+  @property({ type: Boolean })
+  private _coolingDown = false;
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    if (!changedProps.has("active")) {
+      return;
+    }
+
+    const oldActive = changedProps.get("active") as boolean;
```

Comment:
Boolean or undefined for the first update

Reply:
I initialize the property with `false`, so it is never undefined